src/rules/no-restricted-paths.js

```diff
@@ -141,34 +188,34 @@ module.exports = {
 
       matchingZones.forEach((zone, index) => {
         if (!validators[index]) {
-          validators[index] = makePathValidator(zone.from, zone.except);
+          validators[index] = makePathValidators(zone.from, zone.except);
         }
 
-        const {
+        validators[index].forEach(({
           isPathRestricted,
           hasValidExceptions,
           isPathException,
           reportInvalidException,
-        } = validators[index];
-
-        if (!isPathRestricted(absoluteImportPath)) {
-          return;
-        }
-
-        if (!hasValidExceptions) {
-          reportInvalidException(node);
-          return;
-        }
-
-        const pathIsExcepted = isPathException(absoluteImportPath);
-        if (pathIsExcepted) {
-          return;
-        }
-
-        context.report({
-          node,
-          message: `Unexpected path "{{importPath}}" imported in restricted zone.${zone.message ? ` ${zone.message}` : ''}`,
-          data: { importPath },
+        }) => {
+          if (!isPathRestricted(absoluteImportPath)) {
+            return;
+          }
+
+          if (!hasValidExceptions) {
+            reportInvalidException(node);
+            return;
+          }
+
+          const pathIsExcepted = isPathException(absoluteImportPath);
+          if (pathIsExcepted) {
+            return;
+          }
```

Comment:
let's use `.filter` here for all of these, so the part in the forEach is small

Reply:
@ljharb good point 👍🏽 
what do you think about this new version with `.filter` and new functions? https://github.com/import-js/eslint-plugin-import/pull/2466/commits/2d8aad50b618dd7ac63e11cdd3ce871d79cbb582

Reply:
resolving here as it seems to be ok